### DISCUSSION TITLE
Unable to replace an existing file from bucket

### DISCUSSION THREAD

**Discussion by victor-onoja** [supabase/supabase#18540] [Questions]
Please help, I'm uploading a file to a bucket like this:

// 02 - use case upload image

  Future<void> _uploadImage() async {
    final filename = fullname;
    try {
      print('upload image start');
      print(filename);
      print(_image);
      await supabase.storage.from('avatars').upload(
            'public/victory',
            _image!,
            fileOptions: const FileOptions(cacheControl: '3600', upsert: false),
          );
      if (mounted) {
        print('img uploaded');
        ScaffoldMessenger.of(context).showSnackBar(SnackBar(
          content: Text(
            'Profile picture updated successfully :)',
            style:
                responsiveTextStyle(context, 16, Colors.black, FontWeight.bold),
          ),
          backgroundColor: Colors.green,
        ));
      }
    } on PostgrestException catch (error) {
      print('postgres error: ${error.message}');
    } catch (error) {
      print(error);
    }
  }

when i try to update the file with this same code, changing .upload to .update as it is in the docs i get this error:
StorageException(message: The resource was not found, statusCode: 404, error: Not found)

but when i run the code again with .upload i get this error:
StorageException(message: The resource already exists, statusCode: 409, error: Duplicate)  
proving the file exist but i can't update it

my bucket is public, i have insert, select, and update policies enabled on storage.objects

please what can i do?

**Comment by GaryAustin1** [comment]
Do you change upsert to true?
Otherwise points to your upload policy not being met.

**Reply by victor-onoja** [reply]
seems it's working now, changing upsert to true did it i think, and i added policy all to all authenticated users.

the docs make no mention of upsert true. Thanks for the help :)
